.github/workflows/release.yml

```diff
@@ -0,0 +1,236 @@
+name: Release public Lambda layer
+
+on:
+  workflow_dispatch:
+    inputs:
+      layerKind:
+        description: 'Layer kind such as: java-wrapper, java-agent, python3, nodejs'
+        required: true
+      sourceLayerARN:
+        description: 'Clone layer from ARN: (find layer ARN from soaking workflow log)'
+        required: true
+      toLayerName:
+        description: 'Publish to layer name (ie: aws-otel_python38_ver-1-1-0)'
+        required: true
+      aws_region:
+        description: 'Deploy to aws region'
+        retuired: true
+        default: "{\"aws_region\": [ \"us-east-1\", \"us-east-2\", \"us-west-1\", \"us-west-2\", \"ap-south-1\", \"ap-northeast-3\", \"ap-northeast-2\", \"ap-southeast-1\", \"ap-southeast-2\", \"ap-northeast-1\", \"ca-central-1\", \"eu-central-1\", \"eu-west-1\", \"eu-west-2\", \"eu-west-3\", \"eu-north-1\", \"sa-east-1\"]}"
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+    outputs:
+      smoke-matrix-include: ${{ steps.setup.outputs.include }}
+    steps:
+      - name: setup matrix for smoke test
+        id: setup
+        run: |
+          output_json=""
+          case ${{ github.event.inputs.layerKind }} in
+            "python38") output_json="[{\"name\": \"python38\"}, {\"language\": \"python\"}, {\"build_directory\": \"opentelemetry-lambda/python\"}, {\"build_command\": \"cd sample-apps && ./build.sh\"}, {\"terraform_directory\": \"sample-apps/terraform/python/aws-sdk\"}]"
+            ;;
+            "nodejs") output_json="[{\"name\": \"nodejs\"}, {\"language\": \"nodejs\"}, {\"build_directory\": \"opentelemetry-lambda/nodejs\"}, {\"build_command\": \"npm install\"}, {\"terraform_directory\": \"sample-apps/terraform/nodejs/aws-sdk\"}]"
+            ;;
+            "java-agent") output_json="[{\"name\": \"java-agent\"}, {\"language\": \"java\"}, {\"build_directory\": \"java\"}, {\"build_command\": \"./build.sh\"}, {\"terraform_directory\": \"sample-apps/terraform/java/aws-sdk/agent\"}]"
+            ;;
+            "java-wrapper") output_json="[{\"name\": \"java-wrapper\"}, {\"language\": \"java\"}, {\"build_directory\": \"java\"}, {\"build_command\": \"./build.sh\"}, {\"terraform_directory\": \"sample-apps/terraform/java/aws-sdk/wrapper\"}]"
+            ;;
+          esac
+          echo "::set-output name=include::$output_json"
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN }}
+          role-duration-seconds: 1200
+          aws-region: us-east-1
+      - name: Cache layer
+        uses: actions/cache@v2
+        id: cache-layer
```

Comment:
I think it's simpler to download in each region (e.g., merge download and publish step) then to use caching

Reply:
matrix jobs get started at the same time would not hit the cache, though it is not a big problem.